crates/vsock/src/vhu_vsock.rs

```diff
@@ -317,8 +320,30 @@ impl VhostUserBackend<VringRwLock, ()> for VhostUserVsockBackend {
             }
         }
 
-        if device_event != EVT_QUEUE_EVENT && thread.thread_backend.pending_rx() {
-            thread.process_rx(vring_rx, evt_idx)?;
+        if device_event != EVT_QUEUE_EVENT {
+            let mut last_processed_lock = self.last_processed.lock().unwrap();
+            let last_processed = last_processed_lock.deref_mut();
+
+            match last_processed {
+                RxQueueType::Standard => {
+                    if thread.thread_backend.pending_raw_pkts() {
+                        thread.process_raw_pkts(vring_rx, evt_idx)?;
+                        *last_processed = RxQueueType::RawPkts;
+                    }
+                    if thread.thread_backend.pending_rx() {
+                        thread.process_rx(vring_rx, evt_idx)?;
+                    }
+                }
+                RxQueueType::RawPkts => {
+                    if thread.thread_backend.pending_rx() {
+                        thread.process_rx(vring_rx, evt_idx)?;
+                        *last_processed = RxQueueType::Standard;
+                    }
+                    if thread.thread_backend.pending_raw_pkts() {
+                        thread.process_raw_pkts(vring_rx, evt_idx)?;
+                    }
+                }
+            }
```

Comment:
What about moving this in `VhostUserVsockThread`?

I think we can also move `last_processed` there.
For now, we have a single thread that handles all the queue, in the future we can have multiple threads, but I'm expecting that one thread will always handle the same set of queues, so we shouldn't have multiple threads competing for the same queue.